tools/distrib/install_all_python_modules.sh

```diff
@@ -13,45 +13,58 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-# TODO: Integrate this into CI to avoid bitrot.
+set -ex
 
-echo "It's recommended that you run this script from a virtual environment."
+BASEDIR=$(dirname "$0")/../..
+PACKAGES="grpcio_channelz  grpcio_csds  grpcio_admin grpcio_health_checking  grpcio_reflection  grpcio_status  grpcio_testing grpcio_csm_observability grpcio_tests"
+
+# Change to grpc repo root
+cd "$BASEDIR";
 
+# unit-tests setup starts from here
 function maybe_run_command () {
-  if python setup.py --help-commands | grep "$1" &>/dev/null; then
-    python setup.py "$1";
+  if python3 setup.py --help-commands | grep "$1" &>/dev/null; then
+    python3 setup.py "$1";
   fi
 }
 
-set -e
+python3 -m pip install --upgrade "cython<4.0.0rc1";
+python3 setup.py install;
 
-BASEDIR=$(dirname "$0")
-BASEDIR=$(realpath "$BASEDIR")/../..
+# Build and install grpcio_tools
+pushd tools/distrib/python/grpcio_tools;
+  ../make_grpcio_tools.py
+  GRPC_PYTHON_BUILD_WITH_CYTHON=1 pip install .
+popd;
 
-PACKAGES="grpcio_channelz  grpcio_csds  grpcio_admin grpcio_health_checking  grpcio_reflection  grpcio_status  grpcio_testing grpcio_csm_observability grpcio_tests"
+# Build and install grpcio_observability
+pushd src/python/grpcio_observability;
+  ./make_grpcio_observability.py
+  GRPC_PYTHON_BUILD_WITH_CYTHON=1 pip install .
+popd;
+
+# Install xds_protos
+pushd tools/distrib/python/xds_protos;
+  GRPC_PYTHON_BUILD_WITH_CYTHON=1 pip install .
+popd;
+
+# Build and install individual gRPC packages
+pushd src/python;
+  for PACKAGE in ${PACKAGES}; do
+    pushd "${PACKAGE}";
+      python3 setup.py clean;
+      maybe_run_command preprocess
+      maybe_run_command build_package_protos
+      python3 -m pip install .;
+    popd;
+  done
+popd;
+
+# Run tests for grpcio_tests
+pushd src/python/grpcio_tests;
+  python3 setup.py test_lite
+  python3 setup.py test_aio
+  python3 setup.py test_py3_only
+popd;
 
-(cd "$BASEDIR";
-  pip install --upgrade "cython<3.0.0rc1";
-  python setup.py install;
-  pushd tools/distrib/python/grpcio_tools;
-    ../make_grpcio_tools.py
-    GRPC_PYTHON_BUILD_WITH_CYTHON=1 pip install .
-  popd;
-  pushd src/python/grpcio_observability;
-    ./make_grpcio_observability.py
-    GRPC_PYTHON_BUILD_WITH_CYTHON=1 pip install .
-  popd;
-  pushd tools/distrib/python/xds_protos;
-    GRPC_PYTHON_BUILD_WITH_CYTHON=1 pip install .
-  popd;
-  pushd src/python;
-    for PACKAGE in ${PACKAGES}; do
-      pushd "${PACKAGE}";
-        python setup.py clean;
-        maybe_run_command preprocess
-        maybe_run_command build_package_protos
-        python -m pip install .;
-      popd;
-    done
-  popd;
-)
+chmod -R 755 src/
```

Comment:
Why is this necessary? Seems very odd to be modifying permissions on our source code.

Reply:
This was required for internal_ci jobs to pass. It tries to move some files at the end of build and fails due to permissions. This was one of the ways other have resolved this issues as suggested in internal stackoverflow

Reply:
Which files? Are you able to share the failure? Are we generating junk files that shouldn't be under the src repo? This seems like treating the symptom rather than the cause.

Reply:
`
warning: could not open directory 'src/python/grpcio_tests/tests/protoc_plugin/tmpkolcwjrfwkt_test/': Permission denied
warning: could not open directory 'src/python/grpcio_tests/tests/protoc_plugin/tmprgzxk_wwwkt_test/': Permission denied
warning: could not open directory 'src/python/grpcio_tests/tests/protoc_plugin/tmpjtqox7o7wkt_test/': Permission denied
warning: could not open directory 'src/python/grpcio_tests/tests/protoc_plugin/tmpvm0psk_kwkt_test/': Permission denied
`

Reply:
Looks like these are artifacts generated by the test run above. This command would be reasonable to have in a _testing_ script, but not in the developer-oriented script itself. I think if you split out the test like in [this comment](https://github.com/grpc/grpc/pull/37070#discussion_r1773848498) and move it to a separate script, then organizationally, this should be fine.